docs/sdks/uid2-sdk-ref-java.md

```diff
@@ -102,6 +103,183 @@ Available information returned through the SDK is outlined in the following tabl
 | `KeysNotSynced` | The client has failed to synchronize keys from the UID2 service. |
 | `VersionNotSupported` |  The client library does not support the version of the encrypted token. |
 
+-------------------------------------------copy begin-----------------------------------------------
+
+Copy from: https://github.com/IABTechLab/uid2-client-java#usage-for-publishers
+
+## Usage for Publishers
+
+As a publisher, there are two ways to use the UID2 SDK for Java: 
+1. [**Basic Usage**](#basic-usage) is for publishers who want to use this SDK's HTTP implementation (synchronous [OkHttp](https://square.github.io/okhttp/)).
+2. [**Advanced Usage**](#advanced-usage) is for publishers who prefer to use their own HTTP library. 
+
+For an example application that demonstrates both Basic and Advanced usage, see [Java UID2 Integration Example](https://github.com/UnifiedID2/uid2-examples/tree/main/publisher/uid2-java-test-site#readme).
+
+### Basic Usage
+
+If you're using the SDK's HTTP implementation, follow these steps.
+
+1. Create an instance of `PublisherUid2Client` as an instance variable:
+
+   ```java
+   private final PublisherUid2Client publisherUid2Client = new PublisherUid2Client(UID2_BASE_URL, UID2_API_KEY, UID2_SECRET_KEY);
+   ```
+
+2. Call a function that takes the user's email address or phone number as input and generates a `TokenGenerateResponse` object. The following example uses an email address:
+   ```java
+   TokenGenerateResponse tokenGenerateResponse = publisherUid2Client.generateTokenResponse(TokenGenerateInput.fromEmail(emailAddress).doNotGenerateTokensForOptedOut());
+   ```
+
+   :::important
+   <ul><li>Be sure to call this function only when you have obtained legal basis to convert the user’s [directly identifying information (DII)](../ref-info/glossary-uid.md#gl-dii) to UID2 tokens for targeted advertising.</li></ul>
+   
+   <ul><li>Always apply `doNotGenerateTokensForOptedOut()`. This applies `policy=1` in the [/token/generate](../endpoints/post-token-generate.md#token-generation-policy) call. Support for `policy=0` will be removed soon. (**GWH_SW01 to adjust this here + in the repo pending input. Readme line 53**)</li></ul>
+   :::
+
+#### Standard Integration
+
+If you're using standard integration (client and server) (see [UID2 SDK for JavaScript Integration Guide](../guides/publisher-client-side.md)), follow this step: (**GWH_SW02 not sure is this doc link correct, should we link to the client doc, or to both? Same in repo.**)
```

Comment:
use this link

http://localhost:3006/docs/guides/integration-javascript-standard

Reply:
Done, two instances in this file and two instances in the repo readme, thx.